### PR TITLE
Feature/224 http response handler

### PR DESCRIPTION
**Pull request by pedro-lima-jalafund**
[AB#224](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/224)

# Overview

Handling exceptions for 500 and 400 HTTP responses with a custom exception class. Part 1

## Outstanding Tasks

- [ ] 500 HTTP custom message to UI
- [ ]  400 HTTP custom message

Capture

### CODE REVIEW

File: src/api/Jalasoft.TeamUp.Projects.API/local.settings.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "IsEncrypted": false,
   "Values": {
-    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
-    "FUNCTIONS_WORKER_RUNTIME": "dotnet",
+    "AzurJobsStorage": "UseDevelopmentStorage=true",
```
Comment (Raul-Gamarra-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/local.settings.json]: did you mean to change this?

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjectById.cs [hunk 2]
```diff
@@ -28,15 +29,18 @@ public GetProjectById(IProjectsService projectsService)
...
+                return new OkObjectResult(result);
             }
-            else
+            catch (ProjectsException e)
             {
-                return new OkObjectResult(result);
+                return new ContentResult
+                {
+                    StatusCode = e.StatusCode,
+                    Content = e.ProjectsErrorMessage,
```
Comment (paola-quintanilla-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjectById.cs:42]: You need to return an object with this format:
{
  "message" : "message of error"
}

File: src/api/Jalasoft.TeamUp.Projects.API/local.settings.json (hunk 1 above)
Comment (pedro-lima-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/local.settings.json]: my bad, was a typo, sorry, it is fixed now

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs [hunk 3]
```diff
@@ -25,8 +26,19 @@ public GetProjects(IProjectsService projectService)
...
-            return new OkObjectResult(projects);
+            try
+            {
+                var projects = this.projectService.GetProjects();
+                return new OkObjectResult(projects);
+            }
+            catch (ProjectsException e)
+            {
+                return new ContentResult
+                {
```
Comment (paola-quintanilla-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs:37]: You need to return an object with this format:
{
"message" : "message of error"
}

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/PostProject.cs [hunk 4]
```diff
@@ -29,10 +30,29 @@ public PostProject(IProjectsService postProjectService)
...
+                {
+                    StatusCode = e.StatusCode,
+                    Content = e.ProjectsErrorMessage,
+                };
+            }
+            catch (System.Exception)
+            {
+                return new ContentResult
+                {
+                    StatusCode = 500,
```
Comment (paola-quintanilla-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/PostProject.cs:52]: You need to return an object with this format:
{
"message" : "message of error"
}

**Review by Raul-Gamarra-jalafund — APPROVED** [review]

**Review by rodrigo-baldiviezo-jalafund — APPROVED** [review]
LGTM

File: src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsMongoDbRepository.cs [hunk 5]
```diff
@@ -17,7 +16,15 @@ public class ProjectsMongoDbRepository : IProjectsMongoDbRepository
...
         {
             string stringConnection = Environment.GetEnvironmentVariable("MongoSessionServices", EnvironmentVariableTarget.Process);
-            client = new MongoClient(stringConnection);
+            try
+            {
+                client = new MongoClient(stringConnection);
+            }
+            catch (Exception)
+            {
+                throw;
```
Comment (paola-quintanilla-jalafund) [inline src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsMongoDbRepository.cs]: you should manage try catch in core layer

**Review by lina-ramirez-jalafund — APPROVED** [review]

File: src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsMongoDbRepository.cs (hunk 5 above)
Comment (pedro-lima-jalafund) [reply src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsMongoDbRepository.cs]: This was removed

File: Jalasoft.TeamUp.Projects.ProjectsException/ProjectsException.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
```
Comment (paola-quintanilla-jalafund) [inline Jalasoft.TeamUp.Projects.ProjectsException/ProjectsException.cs:1]: using should be inside of namespace
I thing you need to add stylecop to this new project

**Pull request merged**